src/jit/emitarm64.cpp

```diff
@@ -4328,6 +4384,12 @@ void emitter::emitIns_R_R(
             fmt = IF_DV_2P;
             break;
 
+        case INS_ld1:
+        {
+            fmt = IF_LS_2D;
```

Comment:
This should validate reg1, reg2 , size and opt:

assert(isVectorRegister(reg1));
assert(isGeneralRegister(reg2));
 assert(isValidVectorDatasize(size));

assert(isValidArrangement(size, opt));

Reply:
Will do, it seems like we might be overchecking these in some cases though...

We validate things like `isVectorRegister()` and `isIntegerRegister()` at least 4 times (by my count) for any emitted instruction.

Reply:
:-)